hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/kafka/service/KafkaConsumerGroupManager.java

```diff
@@ -0,0 +1,111 @@
+
+    private final Logger logger = LoggerFactory.getLogger(KafkaConsumerGroupManager.class);
+
+    private final BrokerStorage brokerStorage;
+    private final KafkaNamesMapper kafkaNamesMapper;
+    private final Integer partitions;
+    private final String clusterName;
+    private final ExecutorService executor;
+
+    public KafkaConsumerGroupManager(BrokerStorage brokerStorage,
+                                     KafkaNamesMapper kafkaNamesMapper,
+                                     Integer partitions,
+                                     String clusterName) {
+        this.brokerStorage = brokerStorage;
+        this.kafkaNamesMapper = kafkaNamesMapper;
+        this.partitions = partitions;
+        this.clusterName = clusterName;
+        this.executor = Executors.newFixedThreadPool(
+                partitions, new ThreadFactoryBuilder().setNameFormat("kafka-consumer-group-manager-%d").build()
+        );
+    }
+
+    @Override
+    public void createConsumerGroup(Subscription subscription) {
+        IntStream.range(0, partitions).forEach(partition ->
+            CompletableFuture
+                    .runAsync(new CreateConsumerGroupTask(subscription, partition), executor)
+                    .whenComplete((done, e) -> {
+                        if (e == null) {
+                            logger.info("Successfully created consumer group for subscription {} on partition {}, cluster: {}",
+                                    subscription.getQualifiedName(), partition, clusterName);
+                        } else {
+                            logger.error("Failed to create consumer group for subscription {} on partition {}, cluster: {}",
+                                    subscription.getQualifiedName(), partition, clusterName);
+                        }
+                    })
+        );
+    }
+
+    class CreateConsumerGroupTask implements Runnable {
+
+        private final Logger logger = LoggerFactory.getLogger(CreateConsumerGroupTask.class);
+
+        private final Subscription subscription;
+        private final int partition;
+        private final TopicPartition topicPartition;
+
+        CreateConsumerGroupTask(Subscription subscription, int partition) {
+            this.subscription = subscription;
+            this.partition = partition;
+            this.topicPartition = new TopicPartition(subscription.getQualifiedTopicName(), partition);
+        }
+
+        @Override
+        public void run() {
+            logger.info("Creating consumer group for subscription {} on partition {}", subscription.getQualifiedName(), partition);
+
+            int leaderId = brokerStorage.readLeaderForPartition(new TopicAndPartition(topicPartition));
+            ConsumerGroupId groupId = kafkaNamesMapper.toConsumerGroupId(subscription.getQualifiedName());
+            KafkaConsumer<byte[], byte[]> kafkaConsumer = createKafkaConsumer(groupId, leaderId);
```

Comment:
do we have to create a consumer for each partition and connect to a partition leader?
I believe we could operate just as with consumer's module, have a single consumer that connects to the kafka cluster (not by selecting the leader at all), assign all partitions to it, and commit offsets we want to.

Reply:
Yes, it wasn't necessary. I fixed it.